src/mol-model-formats/structure/mol.ts

```diff
@@ -18,13 +19,40 @@ import { IndexPairBonds } from './property/bonds/index-pair';
 import { Trajectory } from '../../mol-model/structure';
 
 export async function getMolModels(mol: MolFile, format: ModelFormat<any> | undefined, ctx: RuntimeContext) {
-    const { atoms, bonds } = mol;
+    const { atoms, bonds, formalCharges } = mol;
 
     const MOL = Column.ofConst('MOL', mol.atoms.count, Column.Schema.str);
     const A = Column.ofConst('A', mol.atoms.count, Column.Schema.str);
     const type_symbol = Column.asArrayColumn(atoms.type_symbol);
     const seq_id = Column.ofConst(1, atoms.count, Column.Schema.int);
 
+    const computedFormalCharges: Column<number> = Column.ofLambda({
+        value: (row: number) => {
+            if (formalCharges) {
+                const numOfCharges = formalCharges.atomIdx.rowCount;
+                /*
+                    Scan the list of formal charges from the properties
+                    block. If one of them has an atom index that matches the
+                    current row return the corresponding charge.
+                */
+                for (let i = 0; i < numOfCharges; i++) {
+                    if ((formalCharges.atomIdx.value(i) - 1) === row) {
+                        return formalCharges.charge.value(i);
+                    }
+                }
+                /*
+                    If the M CHG property is present, every charge in the atom
+                    block is treated as zero.
+                */
+                return 0;
+            }
+            const idx = atoms.formal_charge.value(row).toString();
+            return formalChargeMapper(idx);
+        },
+        rowCount: atoms.formal_charge.rowCount,
+        schema: atoms.formal_charge.schema,
+    });
```

Comment:
Please change this to something like this:

```ts
const computedFormalCharges = new Int32Array(mol.atoms.count);

for (let i = 0; i < mol.atoms.count; i++) {
    computedFormalCharges[i] = atoms.format_charge.value(i);
}

for (let i = 0; i < formalCharges.atomIdx.rowCount; i++) {
    computedFormalCharges[+formalCharges.atomIdx.value(i)] = +formalCharges.charge.value(i)
}

const pdbx_formal_charge = Column.ofIntArray(computedFormalCharges);
```

Reply:
Addressed by https://github.com/molstar/molstar/pull/314/commits/1b431b1d207b5b189fe9d221b7f103b959ab4a68.